### ISSUE TITLE
Make bash more strict

### ISSUE THREAD

**Issue by cjgillot** [jbigot/zpp#3]
Bash has a lot of implicit semantics for non-existent variables.
This can cause competing directives (like $OMP) to be eaten silently.

Setting a stricter mode for bash by default can help a lot.
```
set -e # Abort on first error
set -u # Err on undefined variable
set -o pipefail # Err on redirection failure
```

This would be a breaking change. The zpp standard library is not `set -u` safe.
See gysela merge !633 for a few examples.

**Comment by jbigot** [comment]
I fully agree with `set -u` and `-o pipefail`. Regarding `-e` however, this breaks some perfectly valid patterns like:
```
dosomething
if $?; then ...
```
